ros2pkg/ros2pkg/api/create_api.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import em
+from io import StringIO
+import os
+import sys
+
+
+def expand_template(template_file, data, output_file, minimum_timestamp=None):
+    output = StringIO()
+    interpreter = em.Interpreter(
+        output=output,
+        options={
+            em.BUFFERED_OPT: True,
+            em.RAW_OPT: True,
+        },
+        globals=data,
+    )
+    with open(template_file, 'r') as h:
+        try:
+            interpreter.file(h)
+        except Exception:
+            if os.path.exists(output_file):
+                os.remove(output_file)
+            print("Exception when expanding '%s' into '%s'" %
+                  (template_file, output_file), file=sys.stderr)
+            raise
+    content = output.getvalue()
+    interpreter.shutdown()
```

Comment:
I introduced a `managed interpreter` which calls `shutdown()` in the `__exit__` function. Is that a correct python way of doing?